cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/IOutputFormatter.cs

```diff
@@ -10,12 +10,20 @@ public interface IOutputFormatter
     /// </summary>
     /// <param name="content">The string content to format and write out</param>
     /// <param name="options">The options to use when formatting output</param>
-    void WriteOutput(string content, OutputFormatterOptions options);
+    void WriteOutput(string content, IOutputFormatterOptions options);
 
     /// <summary>
     /// Format and write stream content
     /// </summary>
     /// <param name="content">The stream content to format and write out</param>
     /// <param name="options">The options to use when formatting output</param>
-    void WriteOutput(Stream content, OutputFormatterOptions options);
+    void WriteOutput(Stream content, IOutputFormatterOptions options);
```

Comment:
why is the method still present since we have an async one now?

Reply:
this lib is only meant to be used by the generated code by kiota, it's not meant to be used by "humans" directly (besides maybe implementing variants). A bit like http/serialization/authentication libs. Providing both, when the generated code only needs the async one leads to code duplication and additional maintenance burden for us. As a matter of fact, why do we need the string version as well?

Reply:
That's not strictly true. Users can write custom commands and add those to the CLIs. Like how we have login command

Reply:
even admitting that's a priority we need to care about now, we should then drive them do do the right thing, it's better to use the async method with a stream since it's non-blocking and supports proper cancellation. Adding overload to make the wrong thing to do easier is a dangerous pattern.

Reply:
The contract for the output formatter describes an object that allows taking in content and writing it somewhere. In the case of our default implementations, we write to the console. The `WriteOutput(string content...)` function is useful if a user has string content already and wants to write that. Making them convert the string to a stream then call `WriteOutput(Stream content...)` would cause friction in the experience. It also introduces unnecessary memory overhead because there will be a byte array of ASCII data. I think the synchronous `WriteOutput(Stream content...)` function would be useful in scenarios where a user is in a synchronous context and needs to write a stream. I think assuming they have the option to make their program synchronous will lead to them having a bunch of `Task.Run(...).Wait()` calls in their program which I don't think is much better.

Reply:
After a call with @andrueastman, we agreed on removing the synchronous `WriteOutput(Stream content...)` function.

Reply:
thanks for making those changes. Is there any reason why we still need the string overloads?

Reply:
Yeah. It's for when a user needs to write a string